Unexpected exit code from worker 90 on Wayland

Now espanso was finally working fine on Wayland, but as of yesterday it stopped working. :/

`espanso start` appears to start espanso, but the replacements aren't working. And if I execute `espanso restart`, it says that espanso wasn't running, but it also doesn't start it.

`#acfg#` and `#pacfg#` aren't working either, but the log does say that it quit with an unexpected exit code.

```
& thefish @ hx90 in ~/fromsvngit/espanso (master ✔) 0 [11:32:36]
~ espanso log
11:32:38 [daemon(673798)] [INFO] reading configs from: "/home/thefish/.config/espanso"
11:32:38 [daemon(673798)] [INFO] reading packages from: "/home/thefish/.config/espanso/match/packages"
11:32:38 [daemon(673798)] [INFO] using runtime dir: "/home/thefish/.local/share/espanso"
11:32:38 [daemon(673798)] [WARN] keyboard layout watcher couldn't determine active layout.
11:32:38 [daemon(673798)] [INFO] watching for changes in path: "/home/thefish/.config/espanso"
11:32:38 [daemon(673798)] [INFO] espanso version: 2.1.5-beta
11:32:38 [daemon(673798)] [INFO] spawning the worker process...
11:32:38 [daemon(673798)] [INFO] binded to IPC unix socket: /home/thefish/.local/share/espanso/espansodaemonv2.sock
11:32:38 [worker(673804)] [INFO] reading configs from: "/home/thefish/.config/espanso"
11:32:38 [worker(673804)] [INFO] reading packages from: "/home/thefish/.config/espanso/match/packages"
11:32:38 [worker(673804)] [INFO] using runtime dir: "/home/thefish/.local/share/espanso"
11:32:38 [worker(673804)] [INFO] binded to IPC unix socket: /home/thefish/.local/share/espanso/espansoworkerv2.sock
11:32:38 [worker(673804)] [INFO] monitoring the status of the daemon process
11:32:38 [worker(673804)] [INFO] using WaylandAppInfoProvider
11:32:38 [worker(673804)] [INFO] detection module will use this keyboard layout: [R=evdev, M=pc86, L=us, V=dvorak-intl, O=terminate:ctrl_alt_bksp]
11:32:38 [worker(673804)] [INFO] using EVDEVSource
11:32:38 [worker(673804)] [INFO] Querying modifier status...
11:32:39 [daemon(673798)] [ERROR] received unexpected exit code from worker 90, exiting

& thefish @ hx90 in ~/fromsvngit/espanso (master ✔) 0 [11:32:41]
~
```

OS: Arch Linux
KDE: 5.24
Espanso version: latest git master

The matches file isn't corrupt and I didn't even edit it in the past few weeks.

How can we further investigate this?

I can reproduce this issue. 
```
❯ ./target/release/espanso daemon                                                                              at  19:25:03
19:25:03 [daemon(73121)] [INFO] reading configs from: "/home/ss/.config/espanso"
19:25:03 [daemon(73121)] [INFO] reading packages from: "/home/ss/.config/espanso/match/packages"
19:25:03 [daemon(73121)] [INFO] using runtime dir: "/home/ss/.cache/espanso"
19:25:03 [daemon(73121)] [INFO] system info: Fedora Linux v36 - kernel: 5.18.13-200.fc36.x86_64
19:25:03 [daemon(73121)] [INFO] watching for changes in path: "/home/ss/.config/espanso"
19:25:03 [daemon(73121)] [INFO] espanso version: 2.1.7-beta
19:25:03 [daemon(73121)] [INFO] spawning the worker process...
19:25:03 [daemon(73121)] [INFO] binded to IPC unix socket: /home/ss/.cache/espanso/espansodaemonv2.sock
19:25:03 [worker(73128)] [INFO] reading configs from: "/home/ss/.config/espanso"
19:25:03 [worker(73128)] [INFO] reading packages from: "/home/ss/.config/espanso/match/packages"
19:25:03 [worker(73128)] [INFO] using runtime dir: "/home/ss/.cache/espanso"
19:25:03 [worker(73128)] [INFO] system info: Fedora Linux v36 - kernel: 5.18.13-200.fc36.x86_64
19:25:03 [worker(73128)] [INFO] using WaylandAppInfoProvider
19:25:03 [worker(73128)] [INFO] binded to IPC unix socket: /home/ss/.cache/espanso/espansoworkerv2.sock
19:25:03 [worker(73128)] [WARN] this version of espanso doesn't come with modulo support, so graphical features (such as Forms and Search) might not be available
19:25:03 [worker(73128)] [WARN] EVDEV backend is being used, but without enabling linux capabilities.
19:25:03 [worker(73128)] [WARN]   Although you CAN run espanso EVDEV backend as root, it's not recommended due
19:25:03 [worker(73128)] [WARN]   to security reasons. Espanso supports linux capabilities to limit the attack surface
19:25:03 [worker(73128)] [INFO] monitoring the status of the daemon process
19:25:03 [worker(73128)] [WARN]   area by only leveraging on the CAP_DAC_OVERRIDE capability (needed to work with
19:25:03 [worker(73128)] [WARN]   /dev/input/* devices to detect and inject text) and disabling it as soon as the
19:25:03 [worker(73128)] [WARN]   initial setup is completed.
19:25:03 [worker(73128)] [INFO] detection module will use this keyboard layout: [R=, M=, L=us, V=, O=]
19:25:03 [worker(73128)] [INFO] using EVDEVSource
19:25:03 [worker(73128)] [INFO] Querying modifier status...
[wayland-client error] Attempted to dispatch unknown opcode 0 for wl_shm, aborting.
19:25:03 [daemon(73121)] [ERROR] received unexpected exit code from worker 90, exiting

```
KDE Plasma again, on Fedora Kinoite.

--- a/espanso-detect/Cargo.toml
+++ b/espanso-detect/Cargo.toml
@@ -24,7 +24,7 @@ widestring = "0.4.3"
 [target.'cfg(target_os="linux")'.dependencies]
 libc = "0.2.85"
 scopeguard = "1.1.0"
-sctk = { package = "smithay-client-toolkit", version = "0.14.0", optional = true }
+sctk = { package = "smithay-client-toolkit", version = "0.15.4", optional = true }
 
 [build-dependencies]
 cc = "1.0.73"

